tests/conftest.py

```diff
@@ -44,6 +46,523 @@ def pytest_logger_config(logger_config):
+            f'Failed to create new project {namespace}'
+        )
+
+        instances.append(proj_obj)
+        return proj_obj
+
+    def finalizer():
+        """
+        Delete the Ceph block pool
+        """
+        for instance in instances:
+            if not instance.is_deleted:
+                ocp.switch_to_default_rook_cluster_project()
+                instance.ocp.delete(
+                    resource_name=instance.namespace
+                )
+                instance.ocp.wait_for_delete(
+                    instance.namespace
+                )
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def pvc_factory(request, rbd_storageclass_factory, project_factory):
+    """
+    Create a persistent Volume Claim factory. Calling this fixture creates new
+    PVC. For custom PVC provide 'storageclass' parameter.
+    """
+    instances = []
+    active_project = None
+
+    def factory(
+        project=None,
+        storageclass=None,
+        custom_data=None,
+        status=constants.STATUS_BOUND
+    ):
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+            storageclass (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'StorageClass' kind.
+            custom_data (dict): If provided then PVC object is created
+                by using these data. Parameters `project` and `storageclass`
+                are not used but reference is set if provided.
+            status (str): If provided then factory waits for object to reach
+                desired state.
+
+        Returns:
+            object: helpers.create_pvc instance.
+        """
+        if custom_data:
+            pvc_obj = PVC(**custom_data)
+            pvc_obj.create(do_reload=False)
+        else:
+            nonlocal active_project
+            project = project or active_project or project_factory()
+            active_project = project
+            storageclass = storageclass or rbd_storageclass_factory()
+
+            pvc_obj = helpers.create_pvc(
+                sc_name=storageclass.name,
+                namespace=project.namespace,
+                wait=False
+            )
+            assert pvc_obj, "Failed to create PVC"
+        if status:
+            helpers.wait_for_resource_state(pvc_obj, status)
+        pvc_obj.storageclass = storageclass
+        pvc_obj.project = project
+
+        instances.append(pvc_obj)
+        return pvc_obj
+
+    def finalizer():
+        """
+        Delete the PVC
+        """
+        for instance in instances:
+            if not instance.is_deleted:
+                instance.delete()
+                instance.ocp.wait_for_delete(
+                    instance.name
+                )
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def rbd_pvc_factory(rbd_storageclass_factory, pvc_factory):
```

Comment:
Do I understand correctly that this one returns the same as one above when you use it as I see that in pvc_factory you are also using rbd_storageclass_factory? So maybe it's redundant and if we will document that pvc_factory is by default using rbc storage class we don't need to have this?  Or do I miss something?

Reply:
I think you're right. It might be more clear to users if we only provide one `pvc_factory` and have the returned function take the interface as an argument. This can be propagated to the storageclass_factory as well. I have an example of how that would work here: https://github.com/clacroix12/ocs-ci/commit/80e8a384713fa9e497735a792a096fc4dcd26223

Reply:
Ok, I will remove `cephfs_` and `rbd_` prefixes from all fixtures and make them universal with `interface` parameter set to rbd.